src/components/DocumentUtility/DocumentUtility.tsx

```diff
@@ -24,11 +25,33 @@ export const DocumentUtilityUnStyled = ({
 }: DocumentUtilityProps) => {
   const fileName = getData(document).name;
   const qrcodeUrl = getData(document)?.links?.self?.href ?? "";
+  const logoUrl = getData(document)?.logo;
+  const [imageSettings, setImageSettings] = useState<ImageSettings>();
 
+  if (logoUrl) {
+    const img: HTMLImageElement = new Image();
+    img.src = logoUrl;
+    img.onload = () => {
+      const logoSize = getDimensions({ width: img.width, height: img.height, maxWidth: 100, maxHeight: 100 });
+
+      setImageSettings({
+        src: logoUrl,
+        x: undefined,
+        y: undefined,
+        height: Math.round(logoSize.height),
+        width: Math.round(logoSize.width),
+        excavate: true,
+      });
+    };
+  }
   const qrCodePopover = (url: string) => (
     <Popover id="qr-code-popover" style={{ borderRadius: 0, border: "1px solid #DDDDDD" }}>
-      <Popover.Content style={{ padding: 0 }}>
-        <QRCode bgColor="#FFFFFF" fgColor="#000000" level="Q" style={{ width: 200, padding: "10px" }} value={url} />
+      <Popover.Content style={{ padding: "10px" }}>
+        {logoUrl ? (
+          <QRCode value={url} level="Q" size={200} bgColor="#FFFFFF" fgColor="#000000" imageSettings={imageSettings} />
+        ) : (
+          <QRCode value={url} level="Q" size={200} bgColor="#FFFFFF" fgColor="#000000" />
+        )}
```

Comment:
nice, i din read about the level prop lol, gonna use this on my pr too.

i think you can use `<QRCode value={url} level="Q" size={200} bgColor="#FFFFFF" fgColor="#000000" imageSettings={imageSettings} />` directly without the condition check on `logoUrl`

the reason the ci failed was something else (think ropsten is flaky, the e2e failed while trying to check etherscan address in endorsementChain component), u can check the log by going to github actions and click the respective failing job, it was https://github.com/TradeTrust/tradetrust-website/runs/1198236308?check_suite_focus=true

Reply:
Ohh whoops, I thought it was because of the QRcode for entries without the "logo" attribute. Will make the changes!